box-share-sdk/src/main/java/com/box/androidsdk/share/vm/InviteCollaboratorsVM.java

```diff
@@ -0,0 +1,274 @@
+            BoxCollaboration collaboration = (BoxCollaboration) responses.getResponses().get(0).getResult();
+            if (collaboration.getAccessibleBy() == null) {
+                res[0] = Integer.toString(R.string.box_sharesdk_collaborators_invited);
+            } else {
+                String login = ((BoxUser)(collaboration).getAccessibleBy()).getLogin();
+                res[0] = Integer.toString(R.string.box_sharesdk_collaborator_invited);
+                res[1] = login;
+            }
+
+        } else {
+            res[0] = Integer.toString(R.string.box_sharesdk_collaborators_invited);
+        }
+        return res;
+    }
+    /**
+     * Helper method for processing request if any requests were successful
+     * @param failedCollaboratorsList the list of collaborators for whom requests were not successful
+     * @param name the name of a collaborator that is already added
+     * @param alreadyAddedCount how many collaborators were already added
+     * @return index 0 is string resource code, index 1 is the string formatted part of the message.
+     */
+    @VisibleForTesting
+    static String[] processAddCollabsRequestFailure(List<String> failedCollaboratorsList, String name, int alreadyAddedCount) {
+        String[] res = new String[2];
+        if (!failedCollaboratorsList.isEmpty()) {
+            StringBuilder collaborators = new StringBuilder();
+            for (int i = 0; i < failedCollaboratorsList.size(); i++) {
+                collaborators.append(failedCollaboratorsList.get(i));
+                if (i < failedCollaboratorsList.size() - 1) {
+                    collaborators.append(' ');
+                }
+            }
+            res[0] = Integer.toString(R.string.box_sharesdk_following_collaborators_error);
+            res[1] = collaborators.toString();
+
+        } else if (alreadyAddedCount == 1) {
+            res[0] = Integer.toString(R.string.box_sharesdk_has_already_been_invited);
+            res[1] = name;
+        } else if (alreadyAddedCount > 1) {
+            res[0] = Integer.toString(R.string.box_sharesdk_num_has_already_been_invited);
+            res[1] = Integer.toString(alreadyAddedCount);
+        } else {
+            res[0] = Integer.toString(R.string.box_sharesdk_unable_to_invite);
+        }
+        return res;
+    }
+
+    private static boolean alreadyAddedFailure(String code) {
+        return !SdkUtils.isBlank(code) && code.equals(BoxRequestsShare.AddCollaboration.ERROR_CODE_USER_ALREADY_COLLABORATOR);
+    }
+
+    private static boolean checkIfKnownFailure(BoxResponse<BoxCollaboration> r, HashSet<Integer> failureCodes) {
+        return r.getException() instanceof BoxException && failureCodes.contains(((BoxException) r.getException()).getResponseCode());
+    }
+
+    /**
+     * Generates failure codes for checking known errors
+     * @return a HashSet of known errors
+     */
+    private static HashSet<Integer> generateFailureCodes() {
```

Comment:
This one is a get too. But, looks like it could be in a static initializer

Reply:
I will make the change.